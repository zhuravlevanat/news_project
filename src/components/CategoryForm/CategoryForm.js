import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import React from "react";

const AddCategoryForm = ({newCategory, addNewCategory, addCategoryInList }) => {
    const onFormSubmit = (e) => {
        e.preventDefault();
        addCategoryInList(newCategory);
        addNewCategory('');
    };

    const onChangeCategory = e => addNewCategory(e.target.value);

    return (
        <FormControl component='form' onSubmit={onFormSubmit}>
            <InputLabel htmlFor="my-input">Add your category here...</InputLabel>
            <Input id="my-input" value={newCategory} required onChange={onChangeCategory}/>
            <FormHelperText>We'll try to find latest news for you....</FormHelperText>
        </FormControl>
    );
};

export default AddCategoryForm;
