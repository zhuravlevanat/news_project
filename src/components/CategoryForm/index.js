import {getNewCategory} from "../../redux/Categories/selector";
import {addCategoryInList, addNewCategory} from "../../redux/Categories/actions";
import {connect} from "react-redux";
import AddCategoryForm from "./CategoryForm";

const mapStateToProps = (state) => ({
    newCategory: getNewCategory(state),
});

export const mapDispatchToProps = dispatch => ({
    addNewCategory: payload => dispatch(addNewCategory(payload)),
    addCategoryInList: payload => dispatch(addCategoryInList(payload)),

});

export default connect(mapStateToProps, mapDispatchToProps)(AddCategoryForm);
