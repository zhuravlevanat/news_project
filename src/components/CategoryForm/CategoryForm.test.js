import React from 'react';
import { shallow } from 'enzyme';
import CategoryForm from './CategoryForm';

describe('CategoryForm', () => {
    const mockForm = jest.fn();
    let props = {};
    let form = null;
    beforeEach(() => {
        props = {
            addNewCategory: mockForm,
            addCategoryInList: mockForm,
            newCategory: 'film',
        };
        form = shallow(<CategoryForm {...props} />);
    });
    describe('initial render', () => {
        it('renders properly', () => {
            expect(form).toMatchSnapshot();
        });
        it('should have text field ', () => {
            expect(form).toHaveLength(1);
        });
        it('should have proper props for category input fields', () => {
            expect(form.find('[id="my-input"]').props()).toEqual({
                onChange: expect.any(Function),
                id: 'my-input',
                required: true,
                value: 'film',
            });
        });
    });
    describe('text data in form', () => {
        const labelText = 'Add your category here...';
        const helperText = 'We\'ll try to find latest news for you....';
        it('should equal label text', () => {
            expect(form.find('WithStyles(ForwardRef(InputLabel))').text()).toEqual(labelText);
        });
        it('should equal helper text', () => {
            expect(form.find('WithStyles(ForwardRef(FormHelperText))').text()).toEqual(helperText);
        });
    });

    describe('event handlers', () => {
        it('should set the category value on change event', () => {
            const component = shallow(<CategoryForm {...props} />);
            component.find('[id="my-input"]').simulate('change', {
                target: {
                    value: 'film',
                },
            });
            expect(component.find('[id="my-input"]').prop('value')).toEqual('film');
        });
        it('should call  on change', () => {
            const component = shallow(<CategoryForm {...props} />);
            component.find('WithStyles(ForwardRef(Input))').getElement().props.onChange({ target: { value: 'value' } });
            expect(props.addCategoryInList).toHaveBeenCalledTimes(1);
        });
        it('should call  on submit', () => {
            form.find('[id="my-input"]').simulate('change', {
                target: {
                    value: 'film',
                },
            });
            expect(props.addNewCategory).toHaveBeenCalledTimes(1);
        });
    });
});
