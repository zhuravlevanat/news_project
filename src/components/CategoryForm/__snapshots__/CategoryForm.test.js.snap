// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryForm initial render renders properly 1`] = `ShallowWrapper {}`;

exports[`CategoryForm renders properly 1`] = `ShallowWrapper {}`;
